platform/ios/PolyPodApp/PodApi/PolyOut/FS/PolyOut+FS.swift

```diff
@@ -208,10 +209,12 @@ extension PolyOut {
                 )
                 try FileManager.default.removeItem(at: url)
                 
-                let newUrl = PolyOut.fsPrefix + PolyOut.fsFilesRoot + "/" + newId
+                let newUrl = PolyOut.fsPrefix + PolyOut.fsFilesRoot + "/" + id
                 var fileStore = UserDefaults.standard.value(
                     forKey: PolyOut.fsKey
                 ) as? [String:String?] ?? [:]
+                // TODO: With multiple archives support we can store more than one file in a single fileId.
+                // What should I do about the fileStore?
```

Comment:
@GhenadiePusca. The fileStore can only deal with one filename. But multiple archives there can be multiple filenames stored for on fileId. 

I was thinking of changing fileStore to `[String: [String]]` instead of `[String: String?]`.

What do you think?

Reply:
As far as I understood the changes made with this PR, your suggestion makes sense.

Reply:
I'm not Ghenadie, and maybe I don't get it, but why would we want to store multiple filenames for one field? I was thinking of logic like this:

1. See if `destUrl` was supplied, if not, generate a new UUID (you already have this above)
2. See if `destUrl` already exists, if not, create the directory
3. Extract the archive to `destUrl` - and now there's a design decision: We could overwrite identical files by default, we could throw an error is any file already exists (my preference for now), we could let the caller provide a conflict resolution strategy, even as a callback probably, but I think  this is overkill until we know what use cases we have.

So, however we do (3), I don't see a situation where we have more than a single path on disk per `destUrl` - what's called `newUrl` below.

Reply:
@fhd Things work as you described above. And for (3) we overwrite the files. 

I think the point is another here. The fileStore object is a mapping where we keep track of which archives were extracted to which `destUrl`. And since we are able to extract more than one to the same `destUrl`, we need to modify it to support multiple files.

Reply:
I guess it depends for what it needs to be used. 

Suppose we have one archive `my_facebook_data.zip` which is extracted somewhere in the folder `my_facebook_data`, and the FileStore would contain the `my_facebook_data` id for the destination url. 

Now if `my_facebook_data.zip` is split in, let's say, 3 archives. Then still `my_facebook_data_1.zip`, `my_facebook_data_2.zip`, `my_facebook_data_3.zip` would be extracted in one `my_facebook_data` folder. We can store the all 3 archive names for the same destination, for the means of knowing where a particular archive is unarchived.

But, looking again on the use cases, I am actually not sure if the 3 different names will be ever be used. I mean there is `removeArchive` API, which presumably removes the unarchived content. I don't know if it would be possible/needed to remove the content of a given sub-archive, like `my_facebook_data_2.zip`; as its unarchived content is now mixed with the content of the other 2 archives. So one would actually want to remove the content for all 3 archives in one go, which in this case would be `my_facebook_data` folder.

Reply:
@GhenadiePusca @fhd Correct me if I am wrong, but the fileStore object does not make much sense. Why don't we just read the files at a destUrl from disk. I see that `readDir` already does that for the `url != ""` case. 

Am I missing something?

Reply:
The name is used only one time in `stat` function. 
I think that we should be able to collapse multiple archive names in just one, which makes sense. I mean `my_facebook_data_1.zip`, `my_facebook_data_2.zip`, `my_facebook_data_3.zip` which would be extracted in just one folder, and I think that name of that folder should be stored in `fileStore`.

Reply:
@GhenadiePusca It is also used in `readDir` at the end. And I find the use there pretty confusing. I will add a comment there.

Reply:
Anyway, I will just make `fileStore` support multiple archive names, and then we can figure out how to do things properly when we move this logic to the core.

Reply:
> Well, in the Google case it is not about multiple archives then(if I understand correctly), as those can be perceived as separate/standalone entities, in the same manner as Facebook archive is different from Google archive.

Well, except they need to be included in a single `GoogleAccount`

Reply:
> > Is the intent documented somewhere?
> 
> The intent would be to show the correct/expected file name when `stat` function is called. But I guess it can be anything as it is just metadata.

But I wonder why we have no docs whatsoever in the Swift code base. Is that the usual practice? Not that we have a lot elsewhere; the polyPod, however, is relatively well documented. Going back to the original question, that intent would be documented in the `polyPod` documentation. You can generate it using typedoc (Not totally sure about that...)

Reply:
> Well, except they need to be included in a single `GoogleAccount`

Makes sense, I imagine that Google feature probably would need to be able to read/modify separately each section, which means that each google archive should have different/unique/identifiable destination url, even though those are for the same google account. Based ob the current implementation in this PR, all of the archives are collapsed under the same url, and the info is lost.

Probably the proper usage from js should be:
```
const destUrlMaps = polyOut.importArchive(google_archive_maps, "GoogleAccount")
const destUrlMail = polyOut.importArchive(google_archive_mail, "GoogleAccount")
```
Where the two urls point to the specific archive, unarchived in the "GoogleAccount" folder.

And the archive url should look as such:
```swift
let id = UUID().string
let newUrl = PolyOut.fsPrefix + PolyOut.fsFilesRoot + "/" + destionationURL + "/" + id
```

@timoteipalade what do you think?

Reply:
The files are split by features. See this in line 43:

```
func featureFilesPath() -> URL {
        let documentDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!.resolvingSymlinksInPath()
        return documentDirectory.appendingPathComponent(PolyOut.fsFilesRoot).appendingPathComponent(activeFeature!.id)
    }
    ```

Reply:
Well then, that GoogleAccount folder can be put in the feature directory, still I am not sure if that wrapping in some `GoogleAccount` folder is needed at all.

Reply:
The files are already segregated by feature, as in line 43, where the `activeFeature.id` is appended to the path. 

To me it makes sense that each archive goes into its own folder for google, since they are self contained.

But what about other features where the archives would need to be in the same place? 

The change in this PR would support both. You can choose whether you put all archives in one place, or not.

Reply:
We need to bring up this thing about where we store the data, and how it should be structured in the next weekly dev, and document it somewhere.

Reply:
> Well then, that GoogleAccount folder can be put in the feature directory, still I am not sure if that wrapping in some `GoogleAccount` folder is needed at all.

I don't think it is.

Reply:
Ok, so destinationUrl argument will be only helpful when needing to unarchive at the same location.
 I am somewhat confused, as the trigger for this PR was Google exporting multiple archives, but from the looks of it, it was already supported with the old implementation, or I am completely missing something this Friday afternoon 😄?

Reply:
Say you have 2 archives and no destinationUrl. How would you import them in the same place?

Reply:
I understand, but does Google import need this? As JJ mentioned, each google archive is standalone. 
For Facebook example, when a big zip is split in multiple ones for the means of space, yes, importing to the same location is what is needed, and this PR achieves it.

Reply:
According to the ticket, we want to put all archives in one place: https://jira.polypoly.eu/browse/PROD4POD-1523

Reply:
I struggle to understand the need, as it seems that all of the google archives are standalone, do not need merging, and would need probably separate identification.

Anyway, the PR does the job. Just that in my head, Google import would work also with the old implementation.

Reply:
Yeah, you have a valid point in my opinion. @fhd are we missing something?